worklist/general_utility_functions.py

```diff
@@ -0,0 +1,16 @@
+from collections import defaultdict
+import mwclient
+
+
+# Function that uses MediaWiki API that maps page titles to IDs
+def convert_article_titles_into_ids(titles):
+    page_title_to_id_mapping = defaultdict(int)
+
+    titles_list = ' | '.join(str(page) for page in titles)
```

Comment:
Do we really need the `str` there? If the page title is a unicode string, this might break:

```
>>> str(u'Battle_of_Bạch_Đằng_(938)')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
UnicodeEncodeError: 'ascii' codec can't encode character u'\u1ea1' in position 11: ordinal not in range(128)
```

mwclient seems able to handle unicode strings just fine though:

```
>>> site.api('query', prop='info', titles=u'Battle_of_Bạch_Đằng_(938)')
OrderedDict([(u'batchcomplete', u''), (u'query', OrderedDict([(u'normalized', [OrderedDict([(u'from', u'Battle_of_B\u1ea1ch_\u0110\u1eb1ng_(938)'), (u'to', u'Battle of B\u1ea1ch \u0110\u1eb1ng (938)')])]), (u'pages', OrderedDict([(u'8366427', OrderedDict([(u'pageid', 8366427), (u'ns', 0), (u'title', u'Battle of B\u1ea1ch \u0110\u1eb1ng (938)'), (u'contentmodel', u'wikitext'), (u'pagelanguage', u'en'), (u'pagelanguagehtmlcode', u'en'), (u'pagelanguagedir', u'ltr'), (u'touched', u'2018-05-25T15:53:34Z'), (u'lastrevid', 831374702), (u'length', 14759)]))])), (u'userinfo', OrderedDict([(u'id', 0), (u'name', u'37.228.229.34'), (u'anon', u'')]))]))])
```

By the way, one piece of advice I got a while ago and found extremely helpful: it's usually simpler and saner to use unicode objects "inside" our programs (including using `from __future__ import unicode_literals` all over, which I see you're doing in some modules already), and explicitly encode/decode at the "borders", when consuming or outputting data.

Reply:
![image](https://user-images.githubusercontent.com/24650465/40573974-b1a6ae76-60e7-11e8-846d-32389c50748c.png)

The problem you're mentioning exists only in Python 2. Python 3 strings are in unicode format only. (Example in the screenshot) And since our code is only in Python 3, we need not worry about it.
Hence, we don't need that import and encoding at all. (I've removed those imports now.)

Reply:
Oh, I see. Yeah I assumed Py2 because of the imports, but this should be fine then, thanks for removing them. However, that also means we don't need most (all?) of the explicit calls to `str` in this PR right? I'd suggest just removing them too.

Reply:
I use str method to convert other data types like int into string. Basically, this is to make the code immune to incorrect data types sent as input parameters or given as output by an API.